app/services/layout.ts

```diff
@@ -0,0 +1,76 @@
+import { PersistentStatefulService } from 'services/core/persistent-stateful-service';
+import { mutation } from 'services/core/stateful-service';
+
+export enum ELayout {
+  Default = 'Default',
+}
+
+export enum ELayoutElement {
+  Minifeed = 'Minifeed',
+  LegacyEvents = 'LegacyEvents',
+  Display = 'Display',
+  Mixer = 'Mixer',
+  Scenes = 'Scenes',
+  Sources = 'Sources',
+}
+
+interface ILayoutServiceState {
+  currentLayout: ELayout;
+  slottedWidgets: { [key in ELayoutElement]?: '1' | '2' | '3' | '4' | '5' | '6' };
+  resizes: { bar1: number; bar2: number };
+}
+
+const RESIZE_DEFAULTS = {
+  [ELayout.Default]: {
+    bar1: 156,
+    bar2: 240,
+  },
+};
+
+export class LayoutService extends PersistentStatefulService<ILayoutServiceState> {
+  static defaultState: ILayoutServiceState = {
+    currentLayout: ELayout.Default,
+    slottedWidgets: {
+      Display: '1',
+      Minifeed: '2',
+      Scenes: '3',
+      Sources: '4',
+      Mixer: '5',
+    },
+    resizes: {
+      bar1: 156,
+      bar2: 240,
+    },
+  };
+
+  init() {
+    super.init();
+  }
+
+  setBarResize(bar: 'bar1' | 'bar2', size: number) {
+    this.SET_RESIZE(bar, size);
+  }
+
+  unslottedElements() {
+    return Object.keys(ELayoutElement).filter(
+      el => !this.state.slottedWidgets[el],
+    ) as ELayoutElement[];
+  }
+
+  @mutation()
+  CHANGE_LAYOUT(layout: ELayout) {
+    this.state.currentLayout = layout;
+    this.state.slottedWidgets = {};
+    this.state.resizes = RESIZE_DEFAULTS[layout];
+  }
+
+  @mutation()
+  SLOT_ELEMENT(element: ELayoutElement, slot: '1' | '2' | '3' | '4' | '5' | '6') {
+    this.state.slottedWidgets[element] = slot;
+  }
+
+  @mutation()
+  SET_RESIZE(bar: 'bar1' | 'bar2', size: number) {
```

Comment:
We should be carefully thinking about what we should save in PersistingStatefulsService to avoid ugly migrations in the future. Let try to answer the next questions before saving these values to the persistence state for all layouts.
- Are all layouts are going to have `bar1` and `bar2`. Maybe it's better to not cap the number of bars with this 2
- Is `size` always `vertical` or `horizontal` size? Maybe it's better to use `width` and `height`?
- Should we store settings for each layout separately? For example `SET_RESIZE(layoutName, layoutOptions)`

Reply:
we decided at least for an initial scope that layouts will have either 1 or 2 resize bars. if we add more we won't need to migrate anything and if a layout doesn't have a 2nd bar it should be fine.

size is either width or height depending on the layout, which is why i left it as size.

that's not unreasonable, but it seems like saving a lot of extra data to state for no real reason. the user isn't expected to change layout often or at all once they have a workflow they like